src/sql/src/pure.rs

```diff
@@ -70,47 +71,55 @@ pub async fn purify_create_source(
 
     let mut file = None;
     match connector {
-        CreateSourceConnector::Kafka { broker, topic, .. } => {
-            if !broker.contains(':') {
-                *broker += ":9092";
-            }
+        CreateSourceConnector::Kafka(kafka) => {
+            match kafka {
```

Comment:
I will extract as much of the logic inside the match arm out into the main body of the `CreateSourceConnector::Kafka` arm once the reference stuff is fully plumbed.